sql/hive/src/test/scala/org/apache/spark/sql/sources/hadoopFsRelationSuites.scala

```diff
@@ -510,21 +510,39 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
     }
   }
 
-  // HadoopFsRelation.discoverPartitions() called by refresh(), which will ignore
-  // the given partition data type.
-  ignore("Partition column type casting") {
+  test("Partition column type casting") {
```

Comment:
Do you mind adding a comment beneath this line which reads `// regression test for SPARK-9735` so that readers can quickly figure out what this is supposed to be testing?